### ISSUE TITLE
Linter forcing use of argument in class setter

### ISSUE THREAD

**Issue by nunocastromartins** [eslint/eslint#7351]
**Tell us about your environment**

* **ESLint Version:**
3.7.1
* **Node Version:**
6.8.0
* **npm Version:**
3.10.8

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

```json
{
  "env": {"es6": true, "node": true},
  "extends": "eslint:recommended"
}
```

**What did you do? Please include the actual source code causing the issue.**

```js
exports.Test = class Test {
  set value(v) {
    throw new Error('You must not set the value directly');
  }
};
```

**What did you expect to happen?**
The code should be valid. JavaScript forces a parameter to be declared; the linter should not force us to use it. As far as I can tell object setters behave correctly.

**What actually happened? Please include the actual, raw output from ESLint.**
```
.../test.js
  2:13  error  'v' is defined but never used  no-unused-vars

✖ 1 problem (1 error, 0 warnings)
```

**Comment by not-an-aardvark** [comment]
Thanks for the report. I can reproduce this issue.

The rule currently ignores setter parameters in object literals, because a setter without a parameter is a syntax error. It should also ignore setter parameters in classes.
